.chloggen/3963.yaml

```diff
@@ -0,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. collector, target allocator, auto-instrumentation, opamp, github action)
+component: collector
+
+# A brief description of the change. Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: "Use the env vars provided by the user instead of the ones inferred by the operator"
```

Comment:
Could we be more specific on what is being fixed? 

e.g. Prevent the operator from overriding the env vars defined in the collector CR?